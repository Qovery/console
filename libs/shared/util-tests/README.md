# shared-util-tests

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-tests` to execute the unit tests via [Jest](https://jestjs.io).
