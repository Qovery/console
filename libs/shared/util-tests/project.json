{
  "name": "shared-util-tests",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util-tests/src",
  "projectType": "library",
  "tags": ["scope:shared", "type:util"],
  "implicitDependencies": ["!shared-ui"],
  "targets": {
    "test": {
      "options": {
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "coverage": true
        }
      }
    }
  }
}
