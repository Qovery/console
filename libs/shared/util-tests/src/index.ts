export * from './lib/shared-util-tests'
