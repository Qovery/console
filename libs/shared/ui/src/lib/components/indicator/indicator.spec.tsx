import { renderWithProviders } from '@qovery/shared/util-tests'
import Indicator, { type IndicatorProps } from './indicator'

const props: IndicatorProps = {
  content: <div className="h-3 w-3 rounded-full bg-red-500" />,
}

describe('Indicator', () => {
  it('should match snapshot top end', () => {
    const { baseElement } = renderWithProviders(
      <Indicator side="top" align="end" {...props}>
        <button>Button</button>
      </Indicator>
    )
    expect(baseElement).toMatchSnapshot()
  })
  it('should match snapshot bottom end', () => {
    const { baseElement } = renderWithProviders(
      <Indicator side="bottom" align="end" {...props}>
        <button>Button</button>
      </Indicator>
    )
    expect(baseElement).toMatchSnapshot()
  })
})
