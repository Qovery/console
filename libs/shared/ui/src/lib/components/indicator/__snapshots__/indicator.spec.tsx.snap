// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator should match snapshot bottom end 1`] = `
<body>
  <div>
    <span
      class="relative inline-block"
    >
      <button>
        Button
      </button>
      <span
        class="absolute bottom-0 right-0 translate-x-1/2 translate-y-1/2"
      >
        <div
          class="h-3 w-3 rounded-full bg-red-500"
        />
      </span>
    </span>
  </div>
</body>
`;

exports[`Indicator should match snapshot top end 1`] = `
<body>
  <div>
    <span
      class="relative inline-block"
    >
      <button>
        Button
      </button>
      <span
        class="absolute top-0 right-0 -translate-y-1/2 translate-x-1/2"
      >
        <div
          class="h-3 w-3 rounded-full bg-red-500"
        />
      </span>
    </span>
  </div>
</body>
`;
